--- gst-plugins-bad-1.12.3/ext/openjpeg/gstopenjpeg.h.orig	2017-10-07 08:13:54.124560428 +0700
+++ gst-plugins-bad-1.12.3/ext/openjpeg/gstopenjpeg.h	2017-10-07 08:14:25.968558497 +0700
@@ -38,7 +38,9 @@
 #define OPJ_CPRL CPRL
 #else
 #include <stdio.h>
-# if defined(HAVE_OPENJPEG_2_2)
+# if defined(HAVE_OPENJPEG_2_3)
+#  include <openjpeg-2.3/openjpeg.h>
+# elif defined(HAVE_OPENJPEG_2_2)
 #  include <openjpeg-2.2/openjpeg.h>
 # elif defined(HAVE_OPENJPEG_2_1)
 #  include <openjpeg-2.1/openjpeg.h>
--- gst-plugins-bad-1.12.3/configure.ac.orig	2017-10-07 08:21:10.086533996 +0700
+++ gst-plugins-bad-1.12.3/configure.ac	2017-10-07 08:22:24.875529462 +0700
@@ -2819,6 +2819,11 @@
         AG_GST_PKG_CHECK_MODULES(OPENJPEG_2_2, libopenjp2 >= 2.2)
         if test x"$HAVE_OPENJPEG_2_2" = x"yes"; then
           AC_DEFINE([HAVE_OPENJPEG_2_2], 1, [Define if OpenJPEG 2.2 is used])
+	    dnl include paths changed for v2.3
+            AG_GST_PKG_CHECK_MODULES(OPENJPEG_2_3, libopenjp2 >= 2.3)
+              if test x"$HAVE_OPENJPEG_2_3" = x"yes"; then
+                AC_DEFINE([HAVE_OPENJPEG_2_3], 1, [Define if OpenJPEG 2.3 is used])
+        fi
         fi
     fi
   else
